examples/isometric/level.go

```
@@ -0,0 +1,93 @@
+// Copyright 2021 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build example
+// +build example
+
+package main
+
+import (
+	"log"
+	"math/rand"
+	"time"
+)
+
+// Level represents a Game level.
+type Level struct {
+	w, h int
+
+	tiles    [][]*Tile // 2D array of tiles
+	tileSize int
+}
+
+// Tile returns the tile at the provided coordinates, or nil.
+func (l *Level) Tile(x, y int) *Tile {
+	if x >= 0 && y >= 0 && x < l.w && y < l.h {
+		return l.tiles[x][y]
+	}
+	return nil
+}
+
+// Size returns the size of the Level.
+func (l *Level) Size() (width, height int) {
+	return l.w, l.h
+}
+
+// NewLevel returns a new randomly generated Level.
+func NewLevel() *Level {
+	// Create a 108x108 Level.
+	l := &Level{
+		w:        108,
+		h:        108,
+		tileSize: 64,
+	}
+
+	// Load embedded SpriteSheet.
+	ss, err := LoadSpriteSheet(l.tileSize)
+	if err != nil {
+		log.Fatalf("failed to load embedded spritesheet: %s", err)
```

NewLevel should return an error instead of logging here.